Adding invoice line no picking name 7

eLBati opened the pull request:
moved from https://code.launchpad.net/~agilebg/account-invoicing/adding_invoice_line_no_picking_name_7/+merge/220289

tafaRU committed:
[ADD] module invoice_line_no_picking_name

tafaRU committed:
[FIX] description

tafaRU committed:
[ADD] translation files

tafaRU committed:
[FIX] Updated translation files

tafaRU committed:
[ADD] adding test/invoice_line_no_picking_name.yml

tafaRU committed:
[IMP] test/invoice_line_no_picking_name.yml

tafaRU committed:
[FIX] override of _prepare_invoice_line

tafaRU committed:
[FIX] header of __init__.py

tafaRU committed:
[IMP] Refactoring: replaced picking by invoice

tafaRU committed:
[IMP] Updated the translation files

tafaRU committed:
[REM] key 'active' since not validated using https://github.com/nbessi/openerp-conventions

tafaRU committed:
[FIX] test/invoice_line_no_picking_name.yml

tafaRU committed:
[FIX] test/invoice_line_no_picking_name.yml

tafaRU committed:
[FIX] Added missing security/invoice_security.xml

pedrobaeza commented on invoice_line_no_picking_name/res_config_view.xml:
```diff
@@ -0,0 +1,20 @@
+            </field>
+        </record>
+
+    </data>
```
Please put an empty end line.

pedrobaeza commented on invoice_line_no_picking_name/stock.py:
```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
```
Please name it _stock_picking.py_

pedrobaeza commented on invoice_line_no_picking_name/stock.py:
```diff
@@ -0,0 +1,50 @@
+            invoice_id, invoice_vals, context=context
+        )
+        if move_line:
+            user = self.pool.get('res.users').browse(
```
You can use `self.pool[]` to  shorten the line

pedrobaeza commented on invoice_line_no_picking_name/res_config.py:
```diff
@@ -0,0 +1,36 @@
+
+    _columns = {
+        'group_not_use_picking_name_per_invoice_line': fields.boolean(
+            "Allow not using the picking name on the invoice lines",
```
Better to put a shorter description like 'Not add picking name on invoice lines'

pedrobaeza commented on invoice_line_no_picking_name/res_config.py:
```diff
@@ -0,0 +1,36 @@
+            "Allow not using the picking name on the invoice lines",
+            implied_group="invoice_line_no_picking_name."
+            "group_not_use_picking_name_per_invoice_line",
+            help="Allows you to not use the picking name on the invoice lines."
```
Explain here that the picking name is put when you group pickings. In other case, the result is the same.

pedrobaeza commented on invoice_line_no_picking_name/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+    'version': '0.1',
+    'category': 'Generic Modules/Accounting',
+    'description': """
+This module allows to not use the picking name on the invoice lines.
```
Extend description explaining that the picking name is added when you group some picking in one invoice.

eLBati committed:
[IMP] file names and descriptions

eLBati commented:
Thanks @pedrobaeza , fixed

pedrobaeza commented:
Thanks for the changes. :+1: 

Regards.

eLBati commented:
Maybe I did something wrong.

I rebased this with

```
$ git rebase 7.0 
First, rewinding head to replay your work on top of it...
Applying: [ADD] module invoice_line_no_picking_name
Applying: [FIX] description
Applying: [ADD] translation files
Applying: [FIX] Updated translation files
Applying: [ADD] adding test/invoice_line_no_picking_name.yml
Applying: [IMP] test/invoice_line_no_picking_name.yml
Applying: [FIX] override of _prepare_invoice_line
Applying: [FIX] header of __init__.py
Applying: [IMP] Refactoring: replaced picking by invoice
Applying: [IMP] Updated the translation files
Applying: [REM] key 'active' since not validated using https://github.com/nbessi/openerp-conventions
Applying: [FIX] test/invoice_line_no_picking_name.yml
Applying: [FIX] test/invoice_line_no_picking_name.yml
Applying: [FIX] Added missing security/invoice_security.xml
Applying: [IMP] file names and descriptions
```

having

```
On branch 7.0
Your branch is up-to-date with 'origin/7.0'.
```

and

```
origin  https://github.com/OCA/account-invoicing.git (fetch)
origin  https://github.com/OCA/account-invoicing.git (push)
```

Then

```
$ git status 
On branch adding_invoice_line_no_picking_name_7
Your branch and 'eLBati/adding_invoice_line_no_picking_name_7' have diverged,
and have 56 and 15 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
```

so

```
$ git pull
Merge made by the 'recursive' strategy.
```

and

```
$ git push 
Counting objects: 110, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (93/93), done.
Writing objects: 100% (94/94), 22.97 KiB | 0 bytes/s, done.
Total 94 (delta 57), reused 0 (delta 0)
To git@github.com:eLBati/account-invoicing.git
   5122a59..bbdb544  adding_invoice_line_no_picking_name_7 -> adding_invoice_line_no_picking_name_7
```

But now the diff of this PR is showing other files changed that should not be changed

Is the procedure correct?

pedrobaeza commented:
Firtst of all, you can reset all the mess done with this (I copied your last commit hash):

```
git fetch --all
git reset 5122a59929b3a5618508be0fb407ac3b1271b54f --hard
```

Then, the procedure I follow is this:
- Assure all remotes are up-to-date (same as previous command): `git fetch --all`
- Made the rebase specifying remote and branch: `git rebase origin/7.0`
- Push changes: `git push <your_remote_alias> adding_invoice_line_no_picking_name_7 -f`

Try and tell me.

eLBati commented:
Thanks @pedrobaeza 
So, I missed the `push -f`

joaoalf commented on invoice_line_no_picking_name/res_config.py:
```diff
@@ -0,0 +1,38 @@
+from openerp.osv import fields, osv
+
+
+class stock_config_settings(osv.TransientModel):
```
It's better to use orm.TransientModel instead of osv.TransientModel.
osv is deprecated.

eLBati replied:
Thanks. fixed

max3903 commented:
:+1:

joaoalf commented on invoice_line_no_picking_name/stock_picking.py:
```diff
@@ -0,0 +1,50 @@
+    ):
+        res = super(stock_picking, self)._prepare_invoice_line(
+            cr, uid, group, picking, move_line,
+            invoice_id, invoice_vals, context=context
```
It's usually desirable to initialise context with a dictionary before propagating it:

Like:

``` python
context = context or {}
```

or

``` python
if context is None:
    context = {}
```

eLBati replied:
I usually do not add this when my code does not access to context. In other words, the methods I call should be responsible to handle a None context if they need it, not my method

bwrsandman commented:
:-1: Breaks server tests

eLBati committed:
[FIX] orm instead of osv

eLBati commented:
rebased and red because of https://github.com/OCA/account-invoicing/pull/29

eLBati commented:
green

tafaRU commented:
:+1:

bwrsandman commented:
:+1:
